SwedbankPaySDK/Classes/SwedbankPaySDK+Extensions/MerchantBackendConfiguration.swift

```diff
@@ -0,0 +1,207 @@
+            let session = MerchantBackendConfiguration.makeSession(pinPublicKeys: pinPublicKeys)
+            self.init(session: session, backendUrl: backendUrl, callbackScheme: callbackScheme, headers: headers, domainWhitelist: domainWhitelist, additionalAllowedWebViewRedirects: additionalAllowedWebViewRedirects)
+        }
+        
+        internal init(session: Alamofire.Session, backendUrl: URL, callbackScheme: String?, headers: Dictionary<String, String>?, domainWhitelist: [WhitelistedDomain]?, additionalAllowedWebViewRedirects: [WebViewRedirect]?) {
+            api = MerchantBackendApi(
+                session: session,
+                domainWhitelist: MerchantBackendConfiguration.makeDomainWhitelist(
+                    backendUrl: backendUrl, domainWhitelist: domainWhitelist
+                ),
+                requestDecorator: headers.map(SimpleRequestDecorator.init(headers:))
+            )
+            rootLink = RootLink(href: backendUrl)
+            
+            self.callbackScheme = callbackScheme ?? MerchantBackendConfiguration.getDefaultCallbackScheme()
+            self.additionalAllowedWebViewRedirects = additionalAllowedWebViewRedirects
+        }
+        
+        private static func makeSession(pinPublicKeys: [PinPublicKeys]?) -> Session {
+            if let pinPublicKeys = pinPublicKeys, !pinPublicKeys.isEmpty {
+                var pinEvaluators: [String: PublicKeysTrustEvaluator] = [:]
+                for certificate in pinPublicKeys {
+                    pinEvaluators[certificate.pattern] = PublicKeysTrustEvaluator(
+                        keys: certificate.publicKeys,
+                        performDefaultValidation: true,
+                        validateHost: true
+                    )
+                }
+                return Session(
+                    configuration: URLSessionConfiguration.default,
+                    serverTrustManager: ServerTrustManager(
+                        evaluators: pinEvaluators
+                    )
+                )
+            } else {
+                return Session()
+            }
+        }
+        
+        private static func makeDomainWhitelist(
+            backendUrl: URL,
+            domainWhitelist: [WhitelistedDomain]?
+        ) -> [WhitelistedDomain] {
+            if let domainWhitelist = domainWhitelist, !domainWhitelist.isEmpty {
+                return domainWhitelist
+            } else {
+                let domain = WhitelistedDomain.init(
+                    domain: backendUrl.host,
+                    includeSubdomains: true
+                )
+                return [domain]
+            }
+        }
+        
+        private static func getDefaultCallbackScheme() -> String {
+            let urlTypes = Bundle.main.object(forInfoDictionaryKey: "CFBundleURLTypes") as? [Any]
+            let urlTypeDicts = urlTypes?.lazy.compactMap { $0 as? [AnyHashable : Any] }
```

Comment:
Codacy found an issue: [Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6260585)